### PR TITLE
Added ignore_list_functionality

### PR DESCRIPTION
**Pull request by RohitKochhar**
This PR fixes https://github.com/FairwindsOps/nova/issues/206

## Checklist
* [ x ] I have signed the CLA
* [ x ] I have updated/added any relevant documentation

### What's the goal of this PR?

This PR adds ignore-list functionality when finding helm charts in a cluster. If a user specifies charts they wish for nova to ignore during the finding process, they can do so either by adding either the chart name to `--chart-ignore-list=` or the release name to `--release-ignore-list=`.

### What changes did you make?

I created a filtering function that passes through the found charts in the cluster and compares their chart name and release name attributes to the strings provided by the user, and does not pass them as found charts if there is a match.

### CODE REVIEW

**Comment by RohitKochhar** [comment]
Hey @sudermanjr ,

I was wondering if you could advise how I can test the chart name functionality here, as you can see, my unit tests only cover release names, because I wasn't sure the best way to go about it.

**Comment by sudermanjr** [comment]
@RohitKochhar This looks great! 

For testing the chart ignore list, I think you should be able to construct an input to the test that is a release list, and specify a different chart name, then ignore it.

File: pkg/helm/cluster.go [hunk 1]
```diff
@@ -87,3 +89,32 @@ func (h *Helm) OverrideDesiredVersion(rls *output.ReleaseOutput) {
...
+	for _, release := range deployed {
+		isIgnoredRelease := false
+		isIgnoredChart := false
+		for _, ignoreListedRelease := range releaseIgnoreList {
+			if release.Name == ignoreListedRelease {
+				isIgnoredRelease = true
+			}
+		}
+		for _, ignoreListedChart := range chartIgnoreList {
+			if release.Chart.Name() == ignoreListedChart {
```
Comment (sudermanjr) [inline pkg/helm/cluster.go:109]: In trying to add a test for ignoreChart, I hit a nil pointer exception on this. While we shouldn't have a release that doesn't have a chart, it probably makes sense to check for nil here before calling .Name()

File: pkg/helm/cluster_test.go [hunk 2]
```diff
@@ -82,3 +83,75 @@ func TestHelm_OverrideDesiredVersion(t *testing.T) {
...
+					Name: "foo",
+				},
+				{
+					Name: "bar",
+				},
+			},
+			want: []*release.Release{{
+				Name: "bar",
+			}},
+		},
```
Comment (sudermanjr) [inline pkg/helm/cluster_test.go:147]: Can you please add a test case here for chartIgnoreList? (see previous comments)

**Review by sudermanjr — CHANGES_REQUESTED** [review]

File: pkg/helm/cluster_test.go (hunk 2 above)
Comment (RohitKochhar) [reply pkg/helm/cluster_test.go:148]: Added!

**Comment by RohitKochhar** [comment]
BTW, when updating the documentation (in this [commit](https://github.com/FairwindsOps/nova/pull/252/commits/87cf3bcf837ff05354f5a2aae90d76080601367a)), I built the program off my branch and ran `./nova find -h` and replaced the output originally in `docs/usage.md` with the new output from my program. It seems like a few fields updated, many unrelated to the change made here. Let me know if that's ok.

**Review by bbensky — APPROVED** [review]

**Comment by RohitKochhar** [comment]
Let me know if there is anything else I have to do before this can be merged!

File: pkg/helm/cluster.go [hunk 3]
```diff
@@ -87,3 +89,35 @@ func (h *Helm) OverrideDesiredVersion(rls *output.ReleaseOutput) {
...
+	// Filter out any ignored releases
+	filteredDeployed := make([]*release.Release, len(deployed))
+	filteredCount := 0
+
+	for _, release := range deployed {
+		isIgnoredRelease := false
+		isIgnoredChart := false
+		for _, ignoreListedRelease := range releaseIgnoreList {
+			if release.Name == ignoreListedRelease {
+				isIgnoredRelease = true
```
Comment (rbren) [inline pkg/helm/cluster.go:105]:
```suggestion
				isIgnoredRelease = true
				break
```

File: pkg/helm/cluster.go [hunk 4]
```diff
@@ -87,3 +89,35 @@ func (h *Helm) OverrideDesiredVersion(rls *output.ReleaseOutput) {
...
+			}
+		}
+		for _, ignoreListedChart := range chartIgnoreList {
+			// Check for nil to avoid a potential nil pointer exception
+			if release.Chart != nil {
+				if release.Chart.Name() == ignoreListedChart {
+					isIgnoredChart = true
+				}
+			}
```
Comment (rbren) [inline pkg/helm/cluster.go:114]:
```suggestion
			if release.Chart != nil && release.Chart.Name() == ignoreListedChart {
				isIgnoredChart = true
				break
			}
```

File: pkg/helm/cluster.go [hunk 5]
```diff
@@ -87,3 +89,35 @@ func (h *Helm) OverrideDesiredVersion(rls *output.ReleaseOutput) {
...
+		}
+		if !isIgnoredChart && !isIgnoredRelease {
+			filteredDeployed[filteredCount] = release
+			filteredCount++
```
Comment (rbren) [inline pkg/helm/cluster.go:118]: Rather than tracking the count separately, it'd probably simplify things to just use `append`, even though you don't get the bonus of pre-allocating the slice

```suggestion
			filteredDeployed = append(filteredDeployed, release)
```

**Review by rbren — COMMENTED** [review]
Thanks for the contribution! Just a couple comments on logic

**Comment by RohitKochhar** [comment]
@rbren Thanks for the comments, I updated those and pushed the changes 👍

**Review by rbren — APPROVED** [review]

**Comment by rbren** [comment]
Thanks @RohitKochhar! This is a great improvement

**Comment by RohitKochhar** [comment]
Happy to help, thanks for all your feedback and support in the PR @rbren @sudermanjr @bbensky
